src/SoftCacheTrait.php

```diff
@@ -17,8 +17,12 @@
 	 * @param $output mixed
 	 */
 	public function writeMethodCache($method, array $args, $output) {
-		$cache_key = $this->getCacheKey($args);
-		$this->cache[$method][$cache_key] = $output;
+		try {
+			$cache_key = $this->getCacheKey($args);
```

Comment:
Would be nice to have getCacheKey() having its own try/catch block and return false if exception is thrown.